jax/_src/api.py

```diff
@@ -467,9 +467,100 @@ def get_device_info():
                              cache=_cpp_jit_cache)
   f_jitted = wraps(fun)(cpp_jitted_f)
 
+  f_jitted.lower = partial(_lower,
+                           fun, static_argnums, static_argnames, device,
+                           backend, donate_argnums, inline)
+
   return f_jitted
 
 
+class JitOptions(NamedTuple):
```

Comment:
Perhaps instead of a `NamedTuple` we could use [`TypedDict`](https://www.python.org/dev/peps/pep-0589/)? That has all the same typing benefits, but you can also easily splat it into params.